projects/plugins/jetpack/class.jetpack.php

```diff
@@ -2033,15 +2033,21 @@ public static function load_modules() {
 			$modules = array_diff( $modules, $updated_modules );
 		}
 
+		$is_userless = self::connection()->is_userless();
+
 		foreach ( $modules as $index => $module ) {
-			// If we're in offline mode, disable modules requiring a connection.
-			if ( $is_offline_mode ) {
+			// If we're in offline/user-less mode, disable modules requiring a connection/user connection.
+			if ( $is_offline_mode || $is_userless ) {
```

Comment:
`self::get_module` is a pretty expensive (slow) call. Would it make more sense to check for this in the conditional above with the version_compare, since that check is already firing on every load? Offline mode is pretty uncommon, so adding the user-less check here will increase the calls to get_module.

Reply:
I added memoization to `get_module` in https://github.com/Automattic/jetpack/pull/19068/commits/da7facd030dfb64e2bddfb20fc9e0bd7ed3380c2 (as part of #19068), so it should be fine in that concern.

Do you think it's still worth changing to improve the logic or readability?

Reply:
That's awesome, thank you for doing that! 

There are definitely ways to make this more readable, hence Jeremy's confusion below. The simplest one probably being to separate the offline/userless in their own conditionals, rather than `or` them together.  Or it could get all the modules once at the beginning of the method, and then remove ones that you don't need. Or you could refactor the whole thing and add new methods like `get_offline_modules()` and all that.  Up to you though - not a blocker from me.